lib/config/config.go

```diff
@@ -81,11 +82,15 @@ func ReadXML(r io.Reader, myID protocol.DeviceID) (Configuration, error) {
 	util.SetDefaults(&cfg.Options)
 	util.SetDefaults(&cfg.GUI)
 
-	err := xml.NewDecoder(r).Decode(&cfg)
+	if err := xml.NewDecoder(r).Decode(&cfg); err != nil {
+		return Configuration{}, err
+	}
 	cfg.OriginalVersion = cfg.Version
 
-	cfg.prepare(myID)
-	return cfg, err
+	if err := cfg.prepare(myID); err != nil {
+		return Configuration{}, err
+	}
+	return cfg, nil
```

Comment:
No, I change my mind. Doing that means that prepare() is a method that takes a *Configuration that it modifies, and then it returns the dereferenced `*cfg, nil` - that's ugly and pointless. Better to duplicate the `if`s above. The alternative would be to have it be a `func prepare(cfg Configuration) (Configuration, error)` which is clean but then we should break out the `FolderConfiguration.prepare()` and so on as well, and it's kind of clean that `prepare()` calls `prepare()` on it's children.